topostats/io.py

```diff
@@ -149,16 +159,100 @@ def _load_jpk(self) -> None:
+    #        )
+
+    #    return (images, self._asd_px_to_nm_scaling(scan))
+
+    # uncomment below func and import libasd to use asd formats
+    #def _asd_px_to_nm_scaling(self, scan: Union[libasd.Data2ch_v0, libasd.Data2ch_v1]) -> float:
+    #    """Extracts the pixel to nanometre scaling value from an .asd file object.
+
+    #    Parameters
+    #    ----------
+        scan: Union[libasd.Data2ch_v0, libasd.Data2ch_v1]
+    #        The object resulting from loading the filepath via the libasd library.
+    #    Returns
+    #    -------
+    #    px_to_m_scaling: float
+    #        The length of a single pixel in real length units (nm).
+
+    #    """
+    #    px_to_m_scaling = (
+    #        scan.header.x_scanning_range / scan.header.x_pixel,  # scan range always in nm
+    #        scan.header.y_scanning_range / scan.header.y_pixel,
+    #    )[0]
+    #    return px_to_m_scaling
+
     @staticmethod
     def _extract_jpk(jpk: AFMCreepCompliance) -> np.ndarray:
         """Extract data from jpk object"""
 
     def get_data(self) -> None:
-        """Method to extract image and pixel to nm scaling."""
-        LOGGER.info(f"Extracting image from {self.suffix}")
-        if self.suffix == ".spm":
-            self.image, self.pixel_to_nm_scaling = self.load_spm()
-        if self.suffix == ".jpk":
-            self.image, self.pixel_to_nm_scaling = self.load_jpk()
-        if self.suffix == ".ibw":
-            self.image, self.pixel_to_nm_scaling = self.load_ibw()
+        """Method to extract image, filepath and pixel to nm scaling value, and append these to the
+        img_dic object.
+        """
+        for img_path in self.img_paths:
+            self.img_path = img_path
+            self.filename = img_path.stem
+            self.suffix = img_path.suffix
+            LOGGER.info(f"Extracting image from {self.img_path}")
+            if self.suffix == ".spm":
+                self.image, self.pixel_to_nm_scaling = self.load_spm()
+                self.add_to_dic(self.image, self.img_path.with_name(self.filename), self.pixel_to_nm_scaling)
+            if self.suffix == ".jpk":
+                self.image, self.pixel_to_nm_scaling = self.load_jpk()
+            if self.suffix == ".ibw":
+                self.image, self.pixel_to_nm_scaling = self.load_ibw()
+                self.add_to_dic(self.image, self.img_path.with_name(self.filename), self.pixel_to_nm_scaling)
+            if self.suffix == ".asd":
+                self.img_path = str(self.img_path)
+                self.image, self.pixel_to_nm_scaling = self.load_asd()
+                for i, frame in enumerate(self.image):
+                    pathname = list(Path(self.img_path).parts)
+                    pathname[-1] = self.filename + "_frame_" + str(i)
+                    pathname = Path(*pathname)
+                    self.add_to_dic(frame, pathname, self.pixel_to_nm_scaling)
+
+    def add_to_dic(self, image, img_path, px_2_nm):
```

Comment:
So I'd like to avoid adding `f"{frame}"` in this function due to not all filetypes containing frames and also unpacking the dict in `Filters`, so that the functions can be as easy to use as possible.

The dictionary has now been implemented with docstrings